### PR TITLE
Add task for run jobs

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Shows how to run jobs and monitor them

#### Which issue(s) this PR fixes:

Part of #64

### CODE REVIEW

**Comment by alculquicondor** [comment]
/hold
to prevent premature merge

**Comment by alculquicondor** [comment]
/retest
opened #158 to investigate

File: docs/tasks/README.md [hunk 1]
```diff
@@ -3,6 +3,21 @@
...
+## As a batch administrator
+
+A _batch administrator_ manages the cluster infrastructure and establishes
+quotas and queues.
+
+### [Administer Cluster Quotas](administer_cluster_quotas.md)
+
+Learn how to setup Queues and ClusterQueues to administer quotas in a cluster.
+
+### As a batch user
```
Comment (ahg-g) [inline docs/tasks/README.md:18]: is this supposed to be `##` instead of three?

File: docs/tasks/README.md [hunk 2]
```diff
@@ -3,6 +3,21 @@
...
+
+A _batch user_ runs [workloads](/docs/concepts/queued_workload.md). A typical
+batch user is a researcher, AI/ML engineer, data scientist, among others.
+
+### [Run Jobs](run_jobs.md)
```
Comment (ahg-g) [inline docs/tasks/README.md:23]: why is this a header with nothing underneath it?

File: docs/tasks/run_jobs.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+## 1. Define the Job
+
+Running a Job in Kueue is not much different from [running a Job in a Kubernetes cluster](https://kubernetes.io/docs/tasks/job/)
+without Kueue. The differences are:
+- You should create the Job in a [suspended state](https://kubernetes.io/docs/concepts/workloads/controllers/job/#suspending-a-job),
+  as Kueue will decide when it's the best time to start it.
+- You have to set the Queue you want to submit the Job to. This is done through
+  the annotation `kueue.x-k8s.io/queue-name`.
+- You should include the resource requests and limits for each Job Pod.
```
Comment (ahg-g) [inline docs/tasks/run_jobs.md:40]: why limits is required?

File: docs/tasks/run_jobs.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+## 3. (Optional) Monitor the status of the workload
+
+You can run the following command to see the status of the workload.
+
+```shell
+kubectl -n default describe workload sample-job-sl4bm
+```
+
+If the cluster doesn't have enough quota to run the workload, the output will
```
Comment (ahg-g) [inline docs/tasks/run_jobs.md:103]:
```suggestion
If the ClusterQueue doesn't have enough quota to run the workload, the output will
```

Comment (ahg-g) [reply docs/tasks/run_jobs.md:103]: or 

```suggestion
If there is no enough quota to run the workload, the output will
```

File: docs/tasks/run_jobs.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+    Last Probe Time:       2022-03-28T19:43:03Z
+    Last Transition Time:  2022-03-28T19:43:03Z
+    Message:               workload didn't fit
+    Reason:                Pending
+    Status:                False
+    Type:                  Admitted
+Events:               <none>
+```
+
+After some time, when the cluster has enough quota to run the workload,
```
Comment (ahg-g) [inline docs/tasks/run_jobs.md:128]:
```suggestion
After some time, when there is enough quota to run the workload,
```

File: docs/tasks/run_jobs.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+the output of `kubectl -n default get workloads` will look similar to:
+
+```
+NAME               QUEUE   ADMITTED BY     AGE
+sample-job-sl4bm   main    cluster-total   45s
+```
+
+And when you run `kubectl -n default describe sample-job-sl4bm` again, you will
```
Comment (ahg-g) [inline docs/tasks/run_jobs.md:136]: is this missing `queuedworkloads` after describe?

File: docs/tasks/run_jobs.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+    Last Transition Time:  2022-03-28T19:43:03Z
+    Message:               workload didn't fit
+    Reason:                Pending
+    Status:                False
+    Type:                  Admitted
+Events:               <none>
+```
+
+After some time, when the cluster has enough quota to run the workload,
+the output of `kubectl -n default get workloads` will look similar to:
```
Comment (ahg-g) [inline docs/tasks/run_jobs.md:129]: workloads or queuedworkloads?

File: docs/tasks/README.md (hunk 2 above)
Comment (alculquicondor) [reply docs/tasks/README.md]: I forgot :)

File: docs/tasks/run_jobs.md (hunk 3 above)
Comment (alculquicondor) [reply docs/tasks/run_jobs.md]: Removed

File: docs/tasks/run_jobs.md (hunk 7 above)
Comment (ahg-g) [reply docs/tasks/run_jobs.md:126]: it is fine, I just didn't know that we can use both

**Comment by alculquicondor** [comment]
done

File: docs/tasks/README.md [hunk 8]
```diff
@@ -3,6 +3,23 @@
 This section contains pages that show how to perform individual tasks.
 A task page shows how to do an operation with a short sequence of steps.
```
Comment (cortespao) [inline docs/tasks/README.md:4]: The following pages show you how to perform individual tasks based on two type of roles:
* Batch administrator
* Batch user

File: docs/tasks/README.md [hunk 9]
```diff
@@ -3,6 +3,23 @@
 This section contains pages that show how to perform individual tasks.
```
Comment (cortespao) [inline docs/tasks/README.md:3]: The following section includes the tasks available for the following user personas:

* Batch administrators
* Batch users

File: docs/tasks/README.md [hunk 10]
```diff
@@ -3,6 +3,23 @@
...
 A task page shows how to do an operation with a short sequence of steps.
 
-## [Administer Cluster Resources](administer_cluster_resources.md)
+The tasks are designed to be followed by two personas: _batch administrators_
+and _batch users_.
```
Comment (cortespao) [inline docs/tasks/README.md:8]: The following section includes the tasks available for the following user personas:

* Batch administrators
* Batch users

File: docs/tasks/README.md [hunk 11]
```diff
@@ -3,6 +3,23 @@
...
\ No newline at end of file
+## As a batch administrator
+
+A _batch administrator_ manages the cluster infrastructure and establishes
+quotas and queues.
+
+### [Administer cluster quotas](administer_cluster_quotas.md)
+
+Learn how to setup Queues and ClusterQueues to administer quotas in a cluster.
+
```
Comment (cortespao) [inline docs/tasks/README.md:17]: As a batch administator, learn how to [administer quotas in a cluster](administer_cluster_quotas.md) with `Queues` and `ClusterQueue`.

File: docs/tasks/README.md [hunk 12]
```diff
@@ -3,6 +3,23 @@
...
+## As a batch user
+
+A _batch user_ runs [workloads](/docs/concepts/queued_workload.md). A typical
+batch user is a researcher, AI/ML engineer, data scientist, among others.
+
+### [Run Jobs](run_jobs.md)
+
+Learn how to run a Job on a cluster managed with Kueue.
```
Comment (cortespao) [inline docs/tasks/README.md:25]: As a batch user, learn how to [run a Job on a cluster](run_jobs.md) managed with Kueue.

File: docs/tasks/README.md [hunk 13]
```diff
@@ -3,6 +3,23 @@
...
+and _batch users_.
 
-Learn how to setup Queues and ClusterQueues to administer quotas in a cluster.
\ No newline at end of file
+## As a batch administrator
+
+A _batch administrator_ manages the cluster infrastructure and establishes
+quotas and queues.
+
+### [Administer cluster quotas](administer_cluster_quotas.md)
```
Comment (cortespao) [inline docs/tasks/README.md:14]: You can remove this H3

File: docs/tasks/README.md [hunk 14]
```diff
@@ -3,6 +3,23 @@
...
+
+Learn how to setup Queues and ClusterQueues to administer quotas in a cluster.
+
+## As a batch user
```
Comment (cortespao) [inline docs/tasks/README.md:18]: You can remove this H3

File: docs/tasks/run_jobs.md [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+
+The [ClusterQueue](/docs/concepts/cluster_queue.md) defines the quotas for the
+Queue.
+
+## 1. Define the Job
+
+Running a Job in Kueue is not much different from [running a Job in a Kubernetes cluster](https://kubernetes.io/docs/tasks/job/)
+without Kueue. The differences are:
+- You should create the Job in a [suspended state](https://kubernetes.io/docs/concepts/workloads/controllers/job/#suspending-a-job),
+  as Kueue will decide when it's the best time to start it.
```
Comment (cortespao) [inline docs/tasks/run_jobs.md:37]: Instead of the final "it", could it be "a job"?

File: docs/tasks/run_jobs.md [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+- You have to set the Queue you want to submit the Job to. This is done through
```
Comment (cortespao) [inline docs/tasks/run_jobs.md:38]: ~This is done through~ Use

File: docs/tasks/run_jobs.md [hunk 17]
```diff
@@ -0,0 +1,190 @@
...
+  the annotation `kueue.x-k8s.io/queue-name`.
```
Comment (cortespao) [inline docs/tasks/run_jobs.md:39]: the `kueue.x-k8s.io/queue-name` annotation

File: docs/tasks/run_jobs.md [hunk 18]
```diff
@@ -0,0 +1,190 @@
...
+
+Here is a sample Job with three Pods that just sleep for a few seconds.
+This sample is also available in [config/samples/sample-job.yaml](/config/samples/sample-job.yaml).
+
+```yaml
+# sample-job.yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: sample-job-
```
Comment (cortespao) [inline docs/tasks/run_jobs.md:50]: Is it normal that the name ends with "-"?

File: docs/tasks/run_jobs.md [hunk 19]
```diff
@@ -0,0 +1,190 @@
...
+```
+
+Internally, Kueue will create a corresponding [QueuedWorkload](/docs/concepts/queued_workload.md)
+for this Job with a matching name.
+
+```shell
+kubectl -n default get workloads
+```
+
+The output will be similar to:
```
Comment (cortespao) [inline docs/tasks/run_jobs.md:85]: The output will be similar to the following:

File: docs/tasks/run_jobs.md [hunk 20]
```diff
@@ -0,0 +1,190 @@
...
+
+```
+NAME               QUEUE   ADMITTED BY     AGE
+sample-job-sl4bm   main                    1s
+```
+
+## 3. (Optional) Monitor the status of the workload
+
+You can run the following command to see the status of the workload.
```
Comment (cortespao) [inline docs/tasks/run_jobs.md:94]: You can see the workload status with the following command:

File: docs/tasks/run_jobs.md [hunk 21]
```diff
@@ -0,0 +1,190 @@
...
+
+```shell
+kubectl -n default describe workload sample-job-sl4bm
+```
+
+If the ClusterQueue doesn't have enough quota to run the workload, the output
+will be similar to:
```
Comment (cortespao) [inline docs/tasks/run_jobs.md:101]: will be similar to the following:

File: docs/tasks/run_jobs.md [hunk 22]
```diff
@@ -0,0 +1,190 @@
...
+    Last Transition Time:  2022-03-28T19:43:03Z
+    Message:               workload didn't fit
+    Reason:                Pending
+    Status:                False
+    Type:                  Admitted
+Events:               <none>
+```
+
+After some time, when the ClusterQueue has enough quota to run the workload,
+the output of `kubectl -n default get workloads` will look similar to:
```
Comment (cortespao) [inline docs/tasks/run_jobs.md:126]: When the ClusterQueue has enough quota, the cluster will run the workload.

To see active workloads, run the following command
 `kubectl -n default get workloads` 

The output is similar to the following:

File: docs/tasks/run_jobs.md [hunk 23]
```diff
@@ -0,0 +1,190 @@
...
+
+```
+NAME               QUEUE   ADMITTED BY     AGE
+sample-job-sl4bm   main    cluster-total   45s
+```
+
+And when you run `kubectl -n default describe workload sample-job-sl4bm` again,
+you will observe an event for the workload admission.
+
```
Comment (cortespao) [inline docs/tasks/run_jobs.md:135]: To view an event for the workload admission, run the following command:

 `kubectl -n default describe workload sample-job-sl4bm`

The output is similar to the following:

File: docs/tasks/run_jobs.md [hunk 24]
```diff
@@ -0,0 +1,190 @@
...
+...
+Events:
+  Type    Reason    Age   From           Message
+  ----    ------    ----  ----           -------
+  Normal  Admitted  50s   kueue-manager  Admitted by ClusterQueue cluster-total
+```
+
+Finally, once the workload has finished executing, you will observe a status
+condition.
+
```
Comment (cortespao) [inline docs/tasks/run_jobs.md:146]: What is the command the user must run to see this output?

File: docs/tasks/run_jobs.md [hunk 25]
```diff
@@ -0,0 +1,190 @@
...
+    Last Probe Time:       2022-03-28T19:43:37Z                                                                                                                      
+    Last Transition Time:  2022-03-28T19:43:37Z                                                                                                                      
+    Message:               Job finished successfully                                                                                                                 
+    Reason:                JobFinished                                                                                                                               
+    Status:                True                                                                                                                                      
+    Type:                  Finished
+...
+```
+
+You can find other details in the Job status by running:
```
Comment (cortespao) [inline docs/tasks/run_jobs.md:161]: To review more details about the Job status, run the following command:

File: docs/tasks/run_jobs.md [hunk 26]
```diff
@@ -0,0 +1,190 @@
...
+
+```shell
+kubectl -n default describe job sample-job-sl4bm
+```
+
+The output will be similar to:
```
Comment (cortespao) [inline docs/tasks/run_jobs.md:167]: The output is similar to the following:

**Review by cortespao — COMMENTED** [review]
Editorial review. Please, feel free to disregard any comment that might affect technical completeness.
Happy to discuss if needed 😃

File: docs/tasks/run_jobs.md (hunk 18 above)
Comment (alculquicondor) [reply docs/tasks/run_jobs.md:50]: yes, because the system will add a string at the end, like: "sample-job-abcd"

Comment (cortespao) [reply docs/tasks/run_jobs.md:50]: TIL this! Thanks.

**Pull request merged**